content/client-lib-development-guide/features.textile

```diff
@@ -682,18 +693,17 @@ h4. ChannelOptions
 * @(TB1)@ options provided when instancing a channel, see "Java ChannelOptions":https://github.com/ably/ably-java/blob/master/src/io/ably/types/ChannelOptions.java as a reference
 * @(TB2)@ The attributes of @ChannelOptions@ consist of:
 ** @(TB2a)@ @encrypted@ boolean - when true, enables automatic encryption & decryption of all messages
-** @(TB2b)@ @cipherParams@ @CipherParams@ - when encrypted is true, the cipher params are required
+** @(TB2b)@ @cipherParams@ - This is required if @encrypted@ is true. It consists of either:
+*** @(TB2b1)@ A @CipherParams@ instance, or
+*** @(TB2b2)@ an options hash (or language equivalent) consisting of the key @key@ with value a key. In this case, the client library should call `getDefaultParams`, passing it the key, to obtain a @CipherParams@ instance
```

Comment:
> I question the value of further simplifying it so that you specify the key only as a channel option, because it doesn't really scale well to passing other non-default cipher params.

Is that something we want to support? I'd understood from the skype call that we only wanted to support changing the `key`, not anything else, and wrote this accordingly - I guess I misundestood.

If we want to support an arbitrary subset of cipherParams, then we need to either:

- have `getDefaultParams` be able to take an arbitrary subset of cipherParams and return a complete cipherParams (or throw an exception), so that eg `{encrypted: true, cipherParams: {key: key, mode: 'cbc'}}` becomes just convenient shorthand for `{encrypted: true, cipherParams: Crypto.getDefaultParams({key: key, mode: 'cbc'})}`; or
- specify that `{encrypted: true, cipherParams: {key: key, mode: 'cbc'}}` means calling `getDefaultParams` with the key only, then selectively overwriting what that returns with any non-`key` properties.

I don't really like option (b), it feels a bit weird/inelegant -- and means that we can no longer validate anything in getDefaultParams (eg getDefaultParams can't assert that the keyLength is a valid one for AES if the algorithm might be changed later). So I'd prefer (a). @paddybyers WDYT?

(Though if we do go for (a), then `getDefaultParams` no longer really describes what it's doing. Maybe `getCipherParams`?)